pymc/gp/cov.py

```diff
@@ -179,21 +181,30 @@ def n_dims(self) -> int:
         """The dimensionality of the input, as taken from the
         `active_dims`.
         """
-        # Evaluate lazily in-case this changes.
+        # Evaluate lazily in case this changes.
         return len(self.active_dims)
 
     def _slice(self, X, Xs=None):
         xdims = X.shape[-1]
         if isinstance(xdims, Variable):
-            [xdims] = constant_fold([xdims])
-        if self.input_dim != xdims:
-            warnings.warn(
-                f"Only {self.input_dim} column(s) out of {xdims} are"
-                " being used to compute the covariance function. If this"
-                " is not intended, increase 'input_dim' parameter to"
-                " the number of columns to use. Ignore otherwise.",
-                UserWarning,
-            )
+            try:
+                # For the HSGP.prior_linearized use case, it's OK if X is not constant.
+                [xdims] = constant_fold([xdims], raise_not_constant=True)
```

Comment:
You can change to raise_not_constant=False, no point in having a try except

Reply:
ah ok, done